source/common/http/async_client_impl.cc

```diff
@@ -143,16 +143,30 @@ void AsyncStreamImpl::sendTrailers(HeaderMap& trailers) {
 
 void AsyncStreamImpl::closeLocal(bool end_stream) {
   ASSERT(!(local_closed_ && end_stream));
+  // Due to the fact that send calls can synchronously result in stream closure, it's possible for
```

Comment:
I don't think I follow how this is happening. What's the callstack when we get called after being already closed?

Reply:
I'm in inferring from tests, but I believe this can happen when the connection pool fails to provide a connection, and surfaces this via a locally-created 503 response.

Reply:
But how does that cause `closeLocal()` to get called twice?